### ISSUE TITLE
[BUG] Bokeh trying to use chromedriver when geckodriver installed.

### ISSUE THREAD

**Issue by birdsarah** [bokeh/bokeh#10331]
I have followed the instructions here: https://docs.bokeh.org/en/latest/docs/user_guide/export.html?highlight=export

And conda installed firefox, geckodriver, selenium.

When trying to export_png, I get the following error:

```
~/miniconda3/envs/braid/lib/python3.7/site-packages/bokeh/io/export.py in get_screenshot_as_png(obj, driver, timeout, resources, width, height)
    202         web_driver.maximize_window()
    203         web_driver.get("file:///" + tmp.path)
--> 204         wait_until_render_complete(web_driver, timeout)
    205         [width, height, dpr] = _maximize_viewport(web_driver)
    206         png = web_driver.get_screenshot_as_png()

~/miniconda3/envs/braid/lib/python3.7/site-packages/bokeh/io/export.py in wait_until_render_complete(driver, timeout)
    294     except TimeoutException as e:
    295         _log_console(driver)
--> 296         raise RuntimeError('Bokeh was not loaded in time. Something may have gone wrong.') from e
    297 
    298     driver.execute_script(_WAIT_SCRIPT)

RuntimeError: Bokeh was not loaded in time. Something may have gone wrong.
```

When I run debug, the driver bokeh is trying to use is chromedriver.

![Screenshot from 2020-07-24 18-29-11](https://user-images.githubusercontent.com/1796208/88442849-a0f27e80-cddb-11ea-82b2-5638ab350579.png)

I do also have chromedriver on my path, but not in my conda environment.

After uninstalling chromedriver from my global path, things worked as expected. Not exactly sure what a good solution is here. It's a conflict between my conda environment and the global...

**Comment by mattpap** [comment]
What we could do is to check which browser/driver appears earlier on `$PATH`, and attempt to use drivers in that order. Conda prepends itself to `$PATH`, so that would work with it. Currently we try to detect chromium-based browsers first and then firefox.

**Comment by bryevdv** [comment]
Is this a dupe of https://github.com/bokeh/bokeh/issues/10108?

**Comment by birdsarah** [comment]
> Is this a dupe of #10108?

No. After uninstalling chromedriver everything worked fine. As @mattpap said it's because the codepath currently just tries chromium first rather than using `$PATH` as precedence.

**Comment by bryevdv** [comment]
@mattpap any idea if this is still an issue? There's been quite a bit of refactoring of the webdriver code in the last few years IIRC.

**Comment by mattpap** [comment]
> any idea if this is still an issue? There's been quite a bit of refactoring of the webdriver code in the last few years IIRC.

That's a possibility, but needs to be verified.
